studio/pages/project/[ref]/reports/storage.tsx

```diff
@@ -0,0 +1,126 @@
+import { observer } from 'mobx-react-lite'
+import { NextPageWithLayout } from 'types'
+import { ReportsLayout } from 'components/layouts'
+import {
+  PRESET_CONFIG,
+  REPORTS_DATEPICKER_HELPERS,
+} from 'components/interfaces/Reports/Reports.constants'
+import ReportWidget from 'components/interfaces/Reports/ReportWidget'
+import { queriesFactory } from 'components/interfaces/Reports/Reports.utils'
+import {
+  CacheHitRateChartRenderer,
+  TopCacheMissesRenderer,
+} from 'components/interfaces/Reports/renderers/StorageRenderers'
+import { useEffect, useMemo } from 'react'
+import ReportHeader from 'components/interfaces/Reports/ReportHeader'
+import { DatePickerToFrom, LogsEndpointParams } from 'components/interfaces/Settings/Logs'
+import { useParams } from 'common'
+import ShimmerLine from 'components/ui/ShimmerLine'
+import ReportPadding from 'components/interfaces/Reports/ReportPadding'
+import { useProjectSubscriptionV2Query } from 'data/subscriptions/project-subscription-v2-query'
+import DatePickers from 'components/interfaces/Settings/Logs/Logs.DatePickers'
+
+export const StorageReport: NextPageWithLayout = () => {
+  const { ref: projectRef } = useParams()
+  const report = useStorageReport()
+
+  const { data: subscription } = useProjectSubscriptionV2Query({ projectRef })
+  const plan = subscription?.plan
+
+  const handleDatepickerChange = ({ from, to }: DatePickerToFrom) => {
+    report.mergeParams({
+      iso_timestamp_start: from || '',
+      iso_timestamp_end: to || '',
+    })
+  }
+
+  const datepickerHelpers = useMemo(
+    () =>
+      REPORTS_DATEPICKER_HELPERS.map((helper, index) => ({
+        ...helper,
+        disabled: (index > 0 && plan?.id === 'free') || (index > 1 && plan?.id !== 'pro'),
```

Comment:
we're disabling other than first option if plan is free
and disabling other than first two options if plan is not pro

what about teams/enterprise? cause our pricing page states 28 log retention for teams